prolog/comment.pl

```diff
@@ -1,44 +1,56 @@
-/*
 module(tokenize(comment)
        [comment/2,
         comment_rec/2,
-        comment_token/2,
-        comment_token_rec/2]).
+        comment_token/3,
+        comment_token_rec/3]).
+
+/** <module> Tokenizing comments
+This module defines matchers for comments used by the tokenize module.
+
+@author Stefan Israelsson Tampe 
+@license LGPL v2 or later
+
+Interface Note:
+Start and End matchers is a matcher (-->) that is either evaluated with no 
+extra argument (--> call(StartMatcher)) and it will just match it's token or it 
+can have an extra argument producing the codes matched by the matcher e.g. used
+as --> call(StartMatcher,MatchedCodes). The matchers match start and end codes 
+of the comment, the 2matcher type will represent these matchers. For examples
+see:
+
+  @see tests/test_comments.pl 
+
+The matchers predicates exported nad defined are:
+
+ comment(+Start:2matcher,+End:2matcher)
```

Comment:
1. Why do we want the rules that don't produce the codes in the comment?
2. What is a `2matcher`?  Can you use an existing prolog type for this?